src/shell/notification.js

```diff
@@ -229,84 +230,72 @@ const Source = GObject.registerClass({
         }
 
         // Fix themed icons
-        if (notificationParams.icon) {
+        /* if (notificationParams.icon) {
             let gicon = Gio.Icon.deserialize(notificationParams.icon);
 
             if (gicon instanceof Gio.ThemedIcon) {
                 gicon = getIcon(gicon.names[0]);
                 notificationParams.icon = gicon.serialize();
             }
-        }
+        } */
```

Comment:
This should just be removed if it's not necessary or fixable.

Reply:
Just reimplemented the fix with the new API too :+1: